src/api/procedures/modifyComplianceRequirement.ts

```diff
@@ -62,16 +64,29 @@ export async function prepareModifyComplianceRequirement(
 
   const { conditions: existingConditions } = existingRequirement;
 
-  if (hasSameElements(conditions, existingConditions, conditionsAreEqual)) {
+  if (hasSameElements(newConditions, existingConditions, conditionsAreEqual)) {
     throw new PolymeshError({
       code: ErrorCode.NoDataChange,
       message: 'The supplied condition list is equal to the current one',
     });
   }
 
-  assertRequirementsNotTooComplex(context, conditions, defaultTrustedClaimIssuers.length);
+  const unchangedRequirements = currentRequirements.filter(
```

Comment:
Instead of going through the array twice, you can use lodash `remove` to get `existingRequirement` and then `currentRequirements` will only contain the unchanged ones